### ISSUE TITLE
[core][experimental] Accelerated DAGs should return normal ObjectRefs

### ISSUE THREAD

**Issue by stephanie-wang** [ray-project/ray#43831]
### Description

Accelerated DAGs are an experimental feature to accelerate Ray applications that can be written as static DAGs. Currently, they use a different interface to return the DAG future compared to the ObjectRefs returned by normal Ray tasks. The user must also "release" the DAG output when they are finished with the result, in order to unblock other DAG executions. We should support returning ObjectRefs directly and automatically release the output when the value goes out of scope to improve usability.

**Comment by anyscalesam** [comment]
Let's go with the plan to track this for ADAG Developer Preview; vLLM can proceed. Let's go with Stephanie writing a 1pager design doc and we can see who wants to pick it up.
